core/translate/main_loop.rs

```diff
@@ -992,8 +997,11 @@ pub fn close_loop(
                 program.resolve_label(loop_labels.next, program.offset());
                 match &table.table {
                     Table::BTree(_) => {
-                        let iteration_cursor_id = index_cursor_id.unwrap_or_else(|| {
-                            table_cursor_id.expect("Either index or table cursor must be opened")
+                        let iteration_cursor_id = temp_cursor_id.unwrap_or_else(|| {
+                            index_cursor_id.unwrap_or_else(|| {
+                                table_cursor_id
+                                    .expect("Either index or table cursor must be opened")
```

Comment:
this `.expect()` should be changed since it now includes 3 different possibilities